tensorboard/plugins/hparams/tf_hparams_scatter_plot_matrix_plot/tf-hparams-scatter-plot-matrix-plot.html

```diff
@@ -536,50 +554,31 @@
 
-        const brush = d3.brush()
-              .extent([[-frameMargin + 1, -frameMargin + 1],
-                       [cellWidth - 1 + frameMargin - 1,
-                        cellHeight - 1 + frameMargin - 1]])
-              .on("start", function ([col, metric]) {
-                // Clear the old selection.
-                if (lastBrushedCell !== this) {
-                  if (lastBrushedCell !== null) {
-                    // This will recursively call the 'start', 'brush', and
-                    // 'end' event listeners for the cell with the old selection
-                    // and will update the markers. The outer 'if' above
-                    // prevents infinite recursion.
-                    brush.move(d3.select(lastBrushedCell), null);
-                  }
-                  lastBrushedCell = this;
-                } else {
-                  updateMarkersForBrushSelection(col, metric,
-                                                 d3.brushSelection(this));
-                }
-              })
-              .on("brush", function([col, metric]) {
-                updateMarkersForBrushSelection(col, metric,
-                                               d3.brushSelection(this));
-              })
-              .on("end", function([col, metric]) {
-                updateMarkersForBrushSelection(col, metric,
-                                               d3.brushSelection(this));
-              });
-        cells.append("g").call(brush);
-
-        // Updates marker visibility so that only markers whose
-        // associated session group markers in the cell indexed by 'col, metric'
-        // lie inside 'selection' are shown. The parameter
-        // 'selection' specifies a rectangle in the cell and should be an array
-        // of the form [[x0, y0], [x1, y1]], where (x0,y0) and (x1,y1) are
-        // cell-relative coordinates of the top-left and bottom-right corner of
-        // the rectangle.
-        function updateMarkersForBrushSelection(col, metric, selection) {
-          if (lastBrushSelection === selection) {
-            // Nothing to do if selection hasn't changed.
-            return;
+        // A d3-selection of the cell in 'cells' that has the active
+        // brush selection, or null if the brush is not active.
+        let brushedCellG = null;
+        if (isBrushActive()) {
+          brushedCellG = cells.filter(
+              cellIndex=>_.isEqual(cellIndex, _this._brushedCellIndex));
+          console.assert(brushedCellG.size() == 1,
+                         brushedCellG);
+        }
+
+        // The set of markers (in all cells) that are visible. We keep this
+        // set around so that when the brush selection changes we can change
+        // the "fill" attribute of only the markers we need to. This reduces
+        // the browser's rendering time and makes brushing smoother.
+        let visibleMarkers = new Set(markers.nodes());
```

Comment:
`let visibleMarkers = null` is enough since you always set `visibleMarkers` in L593

Reply:
Darn, line numbers changed and do not know which you are referring to. Since you call `updateVisibleMarkers` in line below, when would `visibleMarkers` be used before assignment?

Reply:
In updateVisibleMarkers 'filterSet(visibleMarkers)' filters the set of visible markers to find only the ones that are not visible and needs change.